src/common/hooks/use-send-form.ts

```diff
@@ -1,29 +1,37 @@
 import { FormikProps } from 'formik';
+
 import { useFetchBalances } from '@common/hooks/account/use-account-info';
 import { useSelectedAsset } from '@common/hooks/use-selected-asset';
 import { usePrevious } from '@stacks/ui';
 import { useAssets } from '@common/hooks/use-assets';
 import React, { useCallback, useEffect } from 'react';
 import { microStxToStx } from '@common/stacks-utils';
-import { FormValues } from '@pages/popup/send';
+import { FormValues } from '@pages/popup/send-form';
 import { removeCommas } from '@common/token-utils';
+import { STX_TRANSFER_TX_SIZE_BYTES } from '@common/constants';
+import { useCurrentFee } from './transaction/use-current-fee';
+import BigNumber from 'bignumber.js';
 
 export function useSendAmountFieldActions({
   setFieldValue,
 }: Pick<FormikProps<FormValues>, 'setFieldValue'>) {
   const balances = useFetchBalances();
   const { selectedAsset, balance } = useSelectedAsset();
+  const fee = useCurrentFee();
   const isStx = selectedAsset?.type === 'stx';
 
   const handleSetSendMax = useCallback(() => {
     if (!selectedAsset || !balances.value) return;
     if (isStx) {
-      const stx = microStxToStx(balances.value.stx.balance);
+      const txFee = microStxToStx(
+        new BigNumber(fee ?? 1).multipliedBy(STX_TRANSFER_TX_SIZE_BYTES).toString()
+      );
+      const stx = microStxToStx(balances.value.stx.balance).minus(txFee);
```

Comment:
Would you be willing to include changes in this PR to take into account locked balance?